tensorboard/plugins/graph/keras_util.py

```diff
@@ -41,18 +40,41 @@
 Please refer to https://github.com/tensorflow/tfjs-layers/blob/master/src/keras_format/model_serialization.ts
 for more complete definition.
 """
+import collections
+import six
+
 from tensorboard.compat.proto.graph_pb2 import GraphDef
 from tensorboard.compat.tensorflow_stub import dtypes
 
 
+# Please refer to below link for more information.
+# https://github.com/tensorflow/tensorflow/blob/1020739e1724346ef0088186f0253defaad3edb6/tensorflow/python/keras/engine/network.py#L1079-L1090
+InboundNodes = collections.namedtuple(
+    'InboundNodes', ['inbound_layer', 'node_index', 'tensor_id', 'kwargs'])
```

Comment:
Honestly, I generally am confused about nature of the fourth data. https://github.com/tensorflow/tensorflow/blob/1020739e1724346ef0088186f0253defaad3edb6/tensorflow/python/keras/engine/network.py#L1086
As you may have noticed, it is set to kwargs and, no, I have yet to find an example.

Reply:
Okay, I found an example:

```python
>>> import tensorflow as tf
>>> inputs = tf.keras.layers.Input(shape=(32, 32, 3))
>>> dropout = tf.keras.layers.Dropout(0.2)(inputs, training=True)
>>> model = tf.keras.models.Model(inputs=inputs, outputs=dropout)
>>> import json
>>> print(json.dumps(json.loads(model.to_json())["config"]["layers"][1], indent=4))
{
    "name": "dropout",
    "class_name": "Dropout",
    "config": {
        "name": "dropout",
        "trainable": true,
        "dtype": "float32",
        "rate": 0.2,
        "noise_shape": null,
        "seed": null
    },
    "inbound_nodes": [
        [
            [
                "input_1",
                0,
                0,
                {
                    "training": true
                }
            ]
        ]
    ]
}
```

As you can see, these `kwargs` are additional arguments passed to the
layer at `call` invocation time rather than layer construction time.
Only certain layers admit extra parameters here.

Reply:
Duh, I by mistake passed it onto the constructor without reading the code carefully. Thanks for the example.